resolwe_bio/tools/expression_rpkum.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+"""Normalize expressions to RPKUM."""
+import argparse
+import gzip
+import math
+
+import pandas as pd
+
+
+def parse_arguments():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Normalize expressions using RPKUM.")
+    parser.add_argument('-c', '--counts', required=True, help="Raw counts file.")
+    parser.add_argument('-m', '--mappability', required=True, help="Mappability file.")
+    parser.add_argument('-o', '--output', required=True, help="Output file name.")
+    return parser.parse_args()
+
+
+def parse_expression_file(exp_file):
+    """Parse expression file to a Pandas dataframe."""
+    with gzip.open(exp_file) as exp:
+        df = pd.read_csv(exp, sep='\t')
+
+        ALLOWED_COLUMNS = ["Gene", "Expression"]
+        if not all(column_label in ALLOWED_COLUMNS for column_label in df.columns.values):
+            print(error('Invalid column headers {} in file {}.'.format(df.columns.values, exp_file)))
+            sys.exit(1)
+
+        # Cast Gene column to string
+        df['Gene'] = df['Gene'].astype('str')
+        # Remove any possible empty rows from the input file
+        df.dropna(inplace=True)
+        df.set_index('Gene', inplace=True)
+
+    return df
+
+
+def parse_mapability_file(mapability_file):
+    """Parse mapability file to a Pandas dataframe."""
+    mapability = pd.read_csv(mapability_file, sep='\t')
+    mapability['gene_id'] = mapability['gene_id'].astype('str')
+    mapability.set_index('gene_id', inplace=True)
+    mapability.dropna(inplace=True)
+
+    return mapability
+
+
+def get_coverage(gene_id, mappability):
+    """Return gene coverage from the mapability file."""
+    return mappability.loc[gene_id]['coverage']
+
+
+def rpkum(gene_id, rc, cov, lib_size):
+    """Compute RPKUM."""
+    if cov == 0:
+        return 0
+    else:
+        return (math.pow(10, 9) * rc) / (lib_size * cov)
+
+def main():
+    """Invoke when run directly as a program."""
+    args = parse_arguments()
+
+    mappability = parse_mapability_file(args.mappability)
+    expression = parse_expression_file(args.counts)
+    lib_size = expression['Expression'].sum()
+
+    # Compute RPKUM expression
+    expression['Expression'] = expression.apply(
```

Comment:
If we read the expression file in a clever way, `expression` is a `pandas.Series` instead of `pandas.DataFrame` because we have just one column (`Expression`) and an index (`Gene`). Then we can use here [pandas.Series.apply](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.apply.html).

Reply:
Accessing index value for the use in the `get_coverage()` gets harder, so there is no clear benefit for changing this to `Series.apply()`. Also, input files are not extremely large, so the performance is not an issue.

Reply:
Replace with
```
    df = pd.concat([expression, mappability], axis=1)
    df['Expression'] = 10 ** 9 * df['Expression'] / df['coverage'] / lib_size
```